Chrysalis-Papilio/Soen490-Papilio-Website
# PAPILIO-91-task

## General

Some of the website components and pages have been created before the backend endpoints associated with them were created. This is creating some discrepancies between what we were expecting and what is already created. To make sure that the backend is called at the right position, we should verify all the endpoints and implement them if they are missing. It would also be a good time to refactor some of the component to reduce duplication.

_Issue number_: Papilio-91

## Task description
- verify getEmployee, getActivity backend connection
- improve test coverage for easier refactoring
- add interaction events on delete employee
- Improve UX on section change in employeeDashboard
- Add automated acceptance test
- Genereal code refactoring

## Manual Test

To test you need to have an employee account with an Admin role. If you don't have one, you can ask me to provide one.
Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)

1. start the backend
`npm run build && npm start`
2. start the website
`npm start`
3. On the main page navigate to the login page
4. Insert the admin account information
5. Once logged in, navigate to the Employee page
6. Verify that you have some employees in your table
7. Click on the action button (add employee, delete employee), they should change to close once clicked
8. Navigate to Activity manager
9. Verify that some activities are in the table
10. Click the add activity button, it should change to close once clicked

## Testing
To run all the unit tests, run the following code: `npm test`
To make sure that the tests have reach the appropriate level of coverage, run the following code: `npm run coverage`
To test the automated acceptance test: (make sure to `npm install` to have cypress installed on your machine)
1. start the server `npm start`
2. start the application `npm start`
3. run `npx cypress run`
